util/webhook/webhook.go

```diff
@@ -342,11 +351,13 @@ func (a *ArgoCDWebhookHandler) Handler(w http.ResponseWriter, r *http.Request) {
 		payload, err = a.bitbucketserver.Parse(r, bitbucketserver.RepositoryReferenceChangedEvent)
 	default:
 		log.Debug("Ignoring unknown webhook event")
+		replyBadRequest(w, "Unknown webhook event")
 		return
 	}
 
 	if err != nil {
 		log.Infof("Webhook processing failed: %s", err)
+		replyBadRequest(w, fmt.Sprintf("Webhook processing failed: %s", err))
```

Comment:
Same as above. And also please use `html.EscapeString(err.Error())` for passing the contents of the error message back to the client.

Reply:
thx, added